packages/@lwc/ssr-runtime/src/reflection.ts

```diff
@@ -16,16 +17,85 @@ import {
 
 import type { LightningElement } from './lightning-element';
 
-// Eventually include globals that also reflect
-const attrsToProps = AriaAttrNameToPropNameMap;
+/**
+ * Map of global attribute or ARIA attribute to the corresponding property name.
+ * Not all global attributes are included, just those from `HTMLElementTheGoodParts`.
+ */
+const attrsToProps = assign(create(null), {
+    accesskey: 'accessKey',
+    dir: 'dir',
+    draggable: 'draggable',
+    hidden: 'hidden',
+    id: 'id',
+    lang: 'lang',
+    spellcheck: 'spellcheck',
+    tabindex: 'tabIndex',
+    title: 'title',
+    ...AriaAttrNameToPropNameMap,
+});
+
+const stringDescriptor = (attrName: string): TypedPropertyDescriptor<string | null> => ({
+    configurable: true,
+    enumerable: true,
+    get(this: LightningElement): string | null {
+        return this.getAttribute(attrName);
+    },
+    set(this: LightningElement, newValue: string | null): void {
+        const currentValue = this.getAttribute(attrName);
+        const normalizedValue = String(newValue);
```

Comment:
```suggestion
        const normalizedValue = toString(newValue);
```

Reply:
Our [`toString`](https://github.com/salesforce/lwc/blob/75ff766c8dabdd28e3308e3f2c8f763c385b056a/packages/%40lwc/shared/src/language.ts#L348-L373) actually does some funky stuff and is mostly designed for pretty-printing error messages. I think we are better off with `String`.

Reply:
The quirks in `toString` are just to avoid the error thrown by `String( [ Object.create( null ) ] )`. Also it stringifies `null` in a weird way because oops. We want `"null"` and don't care if null-prototype objects throw, so I'll stick with `String` here.